pep-0727.rst

```diff
@@ -0,0 +1,516 @@
+to forget to add the docstring for it.
+
+Editors can't check or ensure the consistency of the parameters and their docs.
+
+It's not possible to programatically test these docstrings, for example to ensure
+documentation consistency for the parameters across several similar functions, at
+least not without implementing a custom syntax parser.
+
+Some of these previous formats tried to account for the lack of type annotations
+in older Python versions by including typing information in the docstrings,
+but now that information doesn't need to be in docstrings as there is now an official
+syntax for type annotations.
+
+
+Rationale
+=========
+
+This proposal intends to address these shortcomings by extending and complementing the
+information in docstrings, keeping backwards compatibility with existing docstrings,
+and doing it in a way that leverages the Python language and structure, via type
+annotations with ``Annotated``, and a new function in ``typing``.
+
+The reason why this would belong in the standard Python library instead of an
+external package is because although the implementation would be quite trivial,
+the actual power and benefit from it would come from being a standard, so that
+editors and other tools could implement support for it.
+
+This doesn't deprecate current usage of docstrings, it's transparent to common
+developers (library users), and it's only opt-in for library authors that would
+like to adopt it.
+
+
+Specification
+=============
+
+
+``typing.doc``
+--------------
+
+The main proposal is to have a new function ``doc()`` in the ``typing`` module.
+Even though this is not strictly related to the type annotations, it's expected
+to go in ``Annotated`` type annotations, and to interact with type annotations.
+
+There's also the particular benefit that it could be implemented in the
+``typing_extensions`` package to have support for older versions of Python and
+early adopters of this proposal.
+
+This ``doc()`` function would receive one single parameter ``documentation`` with
+a documentation string.
+
+This string could be a multi-line string, in which case, when extracted by tools,
+should be interpreted cleaning up indentation as if using ``inspect.cleandoc()``,
+the same procedure used for docstrings.
+
+This string could probably contain markup, like Markdown or reST. As that could
+be highly debated, that decision is left for a future proposal, to focus here
+on the main functionality.
+
+This specification targets static analysis tools and editors, and as such, the
+value passed to ``doc()`` should allow static evaluation and analysis. If a
+developer passes as the value something that requires runtime execution
+(e.g. a function call) the behavior of static analysis tools is unspecified
+and they could omit it from their process and results. For static analysis
+tools to be conformant with this specification they need only to support
+statically accessible values.
```

Comment:
It would be nice to be stricter here and say that the same rules for docstrings apply -- if an f-string or string operations are used, tools MUST error or produce a warning.